src/sqlfluff/core/linter/linter.py

```diff
@@ -714,23 +723,34 @@ def render_string(
                     "details."
                 )
             )
+
+        variant_limit = config.get("render_variant_limit")
+        templated_variants: List[TemplatedFile] = []
+        templater_violations: List[SQLTemplaterError] = []
+
         try:
-            templated_file, templater_violations = self.templater.process(
+            for variant, vs in self.templater.process_with_variants(
                 in_str=in_str, fname=fname, config=config, formatter=self.formatter
-            )
+            ):
+                if variant:
+                    templated_variants.append(variant)
+                templater_violations += vs
```

Comment:
does this need to be a set of violations? Could we get repeats if we're doing all variants?

Reply:
yes, we could. I haven't approached the problem of deduping between variants _at all_ yet. I'm going to suggest we don't try and solve that yet - and I'll pick it up as a whole separate challenge at the end, when we can do it for linting too.

I'll add a comment in to make a note on that though, it's a really good point.